app/javascript/pages/home.js

```diff
@@ -15,9 +18,47 @@ const styles = {
     fontSize: '50px'
   }
 };
+
 const Home = () => {
   const navigate = useNavigate();
   const [articles, setArticles] = useState([]);
+  const [isModalOpen, setModalOpen] = useState(false);
+  const [data, setData] = useState({
+    title: '',
+    description: '',
+    creator: ''
+  });
+
+  const handleModalToggle = () => {
+    setModalOpen(!isModalOpen);
```

Comment:
might be better to use:

```suggestion
    setModalOpen(prevValue => !prevValue);
```